### ISSUE TITLE
macos compatibility

### ISSUE THREAD

**Issue by romange** [romange/helio#71]
1. `pmr::monotonic_buffer_resource` does not exist on macos
2. `sys/timerfd.h` does not exist on macos
3. cpu_set_t does not exist on macos but there is this [blog post](http://www.hybridkernel.com/2015/01/18/binding_threads_to_cores_osx.html) that explains how to port them to macos
4. `SOCK_NONBLOCK | SOCK_CLOEXEC` do not exist on macos, instead one should use fcntl with FD_CLOEXEC etc.

**Comment by romange** [comment]
for freebsd install:
```
pkg install cmake gcc autoconf-archive libtool ninja git
pkg install bash
```

**Comment by romange** [comment]
freebsd:
```
CMake Error at cmake/internal.cmake:15 (message):
  Unsupported FreeBSD
Call Stack (most recent call first):
  CMakeLists.txt:28 (include)
```

**Comment by romange** [comment]
@Abhra303 please take a look

**Comment by romange** [comment]
fixed by #108
